apps/shell/app.js

```diff
@@ -23,6 +25,161 @@ if (fs.existsSync('.env')) {
   dotenv.config({path: '.env'});
 }
 
+//Start Terminal Color Scheme Implementation
+//declared constants for directory paths
+const xdg_config_dir = (process.env["XDG_CONFIG_HOME"] || path.join(os.homedir(), ".config"));
+const ood_app_config_root = (process.env["OOD_APP_CONFIG_ROOT"] || '/etc/ood/config/ondemand/apps/shell');
+const userDir = path.join(xdg_config_dir, "ondemand", "apps", "shell", "themes");
+const systemDir = path.join(ood_app_config_root, "themes");
+
+//declared global variables for color schemes
+var schemeObjects;
+
+checkDirSync(userDir);
+checkDirSync(systemDir);
+
+if (process.env["SYSTEM_SCHEMES"] === true) {
+    var user = getSchemeObjects(userDir);
+    var system = getSchemeObjects(systemDir);
+
+   schemeObjects = {...getSchemeObjects(userDir), ...getSchemeObjects(systemDir)};
+
+} else {
+   schemeObjects = {...getSchemeObjects(userDir)};
+}
+
+// helper functions
+function checkDirSync(dir) {  
+  try {
+    fs.statSync(dir);
+  } catch(e) {
+    if (dir === systemDir) {
+      process.env["SYSTEM_SCHEMES"] = false;
+    } else if (dir === userDir) {
+      fs.mkdirSync(dir);
```

Comment:
You can just do this for userDir:

    fs.mkdirSync("/Users/efranz/tmp", {recursive: true})

With `recursive: true` it works like `mkdir -p` and doesn't raise an exception if the directory already exists. But mkdirSync does raise an exception if a valid directory could not be created:

```
> fs.mkdirSync("/Users/efranz/tmp3/foo", {recursive: true})
Thrown:
{ Error: ENOTDIR: not a directory, mkdir '/Users/efranz/tmp3/foo'
    at Object.mkdirSync (fs.js:757:3)
  errno: -20,
  syscall: 'mkdir',
  code: 'ENOTDIR',
  path: '/Users/efranz/tmp3/foo' }
```

in that example tmp3 is a file not a directory.

Better to create the user directory (out of convenience) in one code block, with appropriate try/block, and then read the theme files in a separate block.